### PR TITLE
Dispatch smart-eolp case to smart-org

### PR DESCRIPTION
**Pull request by matsl**
# What

When Hyperbole context in org-mode should not be active, drop to later
handlers in hkey-alist i.e. possibly smart-org handler.

# Why

When we say that a nil value `hsys-org-enable-smart-keys` make M-RET
behave as it normally does is suggests that the smart-eol function
should not be active in org mode. Or!?

# Note

Issue comes from report in #603.

### CODE REVIEW

File: ChangeLog [hunk 1]
```diff
@@ -1,3 +1,11 @@
+2024-11-16  Mats Lidell  <matsl@gnu.org>
+
+* test/hsys-org-tests.el (hsys-org--meta-return-on-end-of-line): Add test
+    case for eol action in org-mode.
+
+* hui-mouse.el (hkey-alist): Guard action-key-eol-function from acting
+    when we should be delegate to org-mode.
```
Comment (rswgnu) [inline ChangeLog:7]: ‘should delegate’, no ‘be’.

File: test/hsys-org-tests.el [hunk 2]
```diff
@@ -225,6 +225,58 @@ This is independent of the setting of `hsys-org-enable-smart-keys'."
...
+        (goto-char 1)
+        (end-of-line)
+        (with-mock
+          (mock (hsys-org-meta-return) => t)
+          (should (equal hsys-org-enable-smart-keys v)) ; Ert traceability
+          (should (action-key))))))
+  (let ((hsys-org-enable-smart-keys t)
+        (v t))
+    ;;; One line no return
+    ;; At end of line is also end of file so smart-eolp filters out
```
Comment (matsl) [inline test/hsys-org-tests.el:255]: @rswgnu Any thoughts about this corner case?

**Comment by matsl** [comment]
> Since with the mouse, you can select our points in the window beyond the end of the buffer and expect that to set the buffer to point-max, in this case, we should treat EOL as EOB, that is prioritizing the end of buffer over end of line, whatever that means for the code.

Yes, I understand that but the behavior is likely confusing for the user. Not sure what is the best solution. We could signal an error if at EOB and hsys-org-enable-smart-keys=t. 

Anyway, that is for another PR if we want to fix that. Thanks for approving. It has been merged and pushed.
